### DISCUSSION TITLE
Unable to run test_joint_trajectory_controller ROS2 Rolling

### DISCUSSION THREAD

**Discussion by schavarriaiii** [UniversalRobots/Universal_Robots_ROS2_Driver#841] [Q&A]
I'm using the ROS2 Rolling version of the UR robot driver with a ur3e arm. I am trying to test controlling the movement of the arm, and am encountering an issue regarding one of the robot joints being out of the starting configuration limits, the command and output is pasted below.

ros2 launch ur_robot_driver test_joint_trajectory_controller.launch.py
[INFO] [launch]: All log files can be found below /home/nucleari/.ros/log/2023-10-09-18-34-09-368321-nucleari-GP66-Leopard-11UG-10172
[INFO] [launch]: Default logging verbosity is set to INFO
[INFO] [publisher_joint_trajectory_controller-1]: process started with pid [10175]
[publisher_joint_trajectory_controller-1] [INFO] [1696890849.728106417] [publisher_joint_trajectory_controller]: Goal "pos1" has definition trajectory_msgs.msg.JointTrajectoryPoint(positions=[0.785, -1.57, 0.785, 0.785, 0.785, 0.785], velocities=[], accelerations=[], effort=[], time_from_start=builtin_interfaces.msg.Duration(sec=4, nanosec=0))
[publisher_joint_trajectory_controller-1] [INFO] [1696890849.729072963] [publisher_joint_trajectory_controller]: Goal "pos2" has definition trajectory_msgs.msg.JointTrajectoryPoint(positions=[0.0, -1.57, 0.0, 0.0, 0.0, 0.0], velocities=[], accelerations=[], effort=[], time_from_start=builtin_interfaces.msg.Duration(sec=4, nanosec=0))
[publisher_joint_trajectory_controller-1] [INFO] [1696890849.729702787] [publisher_joint_trajectory_controller]: Goal "pos3" has definition trajectory_msgs.msg.JointTrajectoryPoint(positions=[0.0, -1.57, 0.0, 0.0, -0.785, 0.0], velocities=[], accelerations=[], effort=[], time_from_start=builtin_interfaces.msg.Duration(sec=4, nanosec=0))
[publisher_joint_trajectory_controller-1] [INFO] [1696890849.730314294] [publisher_joint_trajectory_controller]: Goal "pos4" has definition trajectory_msgs.msg.JointTrajectoryPoint(positions=[0.0, -1.57, 0.0, 0.0, 0.0, 0.0], velocities=[], accelerations=[], effort=[], time_from_start=builtin_interfaces.msg.Duration(sec=4, nanosec=0))
[publisher_joint_trajectory_controller-1] [INFO] [1696890849.730493884] [publisher_joint_trajectory_controller]: Publishing 4 goals on topic "/joint_trajectory_controller/joint_trajectory" every {wait_sec_between_publish} s
[publisher_joint_trajectory_controller-1] [WARN] [1696890850.659552023] [publisher_joint_trajectory_controller]: Starting point limits exceeded for joint wrist_1_joint !
[publisher_joint_trajectory_controller-1] [WARN] [1696890855.732244437] [publisher_joint_trajectory_controller]: Start configuration is not within configured limits!
[publisher_joint_trajectory_controller-1] [WARN] [1696890861.732180061] [publisher_joint_trajectory_controller]: Start configuration is not within configured limits!

After seeing this, I went to the test_goal_publishers_config.yaml file (located in /ur_robot_driver/config and attempted to edit the starting point limits, and move the actual robot into a valid starting position in order to avoid this error. No matter what the configuration of the arm and the settings in the file (which is copied below) I could not get this error message to go away. Any help would be appreciated. I also attempted to set the check_starting_point: false and that did not change anything.

Edited test_goal_publishers_config.yaml file:

publisher_scaled_joint_trajectory_controller:
  ros__parameters:

	controller_name: "scaled_joint_trajectory_controller"
	wait_sec_between_publish: 6

	goal_names: ["pos1", "pos2", "pos3", "pos4"]
	pos1:
  	positions: [0.785, -1.57, 0.785, 0.785, 0.785, 0.785]
	pos2:
  	positions: [0.0, -1.57, 0.0, 0.0, 0.0, 0.0]
	pos3:
  	positions: [0.0, -1.57, 0.0, 0.0, -0.785, 0.0]
	pos4:
  	positions: [0.0, -1.57, 0.0, 0.0, 0.0, 0.0]

	joints:
  	- shoulder_pan_joint
  	- shoulder_lift_joint
  	- elbow_joint
  	- wrist_1_joint
  	- wrist_2_joint
  	- wrist_3_joint

	check_starting_point: true
	starting_point_limits:
  	shoulder_pan_joint: [-3,3]
  	shoulder_lift_joint: [-3,3]
  	elbow_joint: [-3,3]
  	wrist_1_joint: [-3,3]
  	wrist_2_joint: [-3,3]
  	wrist_3_joint: [-3,3]

publisher_joint_trajectory_controller:
  ros__parameters:

	controller_name: "joint_trajectory_controller"
	wait_sec_between_publish: 6

	goal_names: ["pos1", "pos2", "pos3", "pos4"]
	pos1:
  	positions: [0.785, -1.57, 0.785, 0.785, 0.785, 0.785]
	pos2:
  	positions: [0.0, -1.57, 0.0, 0.0, 0.0, 0.0]
	pos3:
  	positions: [0.0, -1.57, 0.0, 0.0, -0.785, 0.0]
	pos4:
  	positions: [0.0, -1.57, 0.0, 0.0, 0.0, 0.0]

	joints:
  	- shoulder_pan_joint
  	- shoulder_lift_joint
  	- elbow_joint
  	- wrist_1_joint
  	- wrist_2_joint
  	- wrist_3_joint

	check_starting_point: true
	starting_point_limits:
  	shoulder_pan_joint: [-3,3]
  	shoulder_lift_joint: [-3,3]
  	elbow_joint: [-3,3]
  	wrist_1_joint: [-3,3]
  	wrist_2_joint: [-3,3]
  	wrist_3_joint: [-3,3]

**Comment by fmauch** [comment]
Well, if you are sure what you are doing, then you could also change the `check_starting_point: true` to `check_starting_point: false`. That's what I usually do when I test in simulation. With a real robot, make sure that the robot is actually in a joint configuration close to the first trajectory point, as the robot will try to reach that within 4 seconds.

Other than that, the files you posted seem not to be correct yaml files, but I assume that is a copy & paste error.

There's not much more I can say to this, as I have experience this feature to be working well so far...

**Reply by schavarriaiii** [reply]
Actually we just managed to get this to run when moving the robot closer to the first trajectory point. Thanks for the input
